src/circllhist.c

```diff
@@ -512,6 +512,84 @@ hist_approx_sum(const histogram_t *hist) {
   return sum;
 }
 
+double
+hist_approx_stddev(const histogram_t *hist) {
+  int i;
+  double total_count = 0.0;
+  double s1 = 0.0;
+  double s2 = 0.0;
+  ASSERT_GOOD_HIST(hist);
+  if(hist->used == 0) return 0.0;
+  for(i=0; i<hist->used; i++) {
+    if(hist_bucket_isnan(hist->bvs[i].bucket)) continue;
+    double midpoint = hist_bucket_midpoint(hist->bvs[i].bucket);
+    double count = hist->bvs[i].count;
+    total_count += count;
+    s1 += midpoint * count;
+    s2 += pow(midpoint, 2.0) * count;
+  }
```

Comment:
total_count can legitimately be 0 here if the bucket is a NaN bucket.  We should either return 0 or NaN... (ask @HeinrichHartmann which)